go/test_benchmark.sh

```
@@ -0,0 +1,159 @@
+echo "=== S3 Upload Benchmark Results ===" > "$RESULTS_FILE"
+echo "Date: $(date)" >> "$RESULTS_FILE"
+echo "Bucket: $BUCKET" >> "$RESULTS_FILE"
+echo "" >> "$RESULTS_FILE"
+
+# 1. 小さいファイルのディレクトリアップロード比較
+echo -e "${CYAN}1. Small Files Directory Upload (100 files, ~7KB each)${NC}"
+log_result "1. Small Files Directory Upload"
+log_result "Directory: benchmark/small (Total: 704KB)"
+echo ""
+
+cd "$GO_DIR"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)